src/main/java/com/cegeka/academy/web/rest/ChallengeController.java

```diff
@@ -1,7 +1,22 @@
 package com.cegeka.academy.web.rest;
 
+import com.cegeka.academy.domain.Challenge;
+import com.cegeka.academy.service.challenge.ChallengeServiceImp;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 
 @Controller
+@RequestMapping("/challenge")
 public class ChallengeController {
+
+    @Autowired
+    ChallengeServiceImp challengeService;
+
+    @PostMapping("/create")
+    public String saveChallenge(@RequestBody Challenge challenge) {
+        return challengeService.saveChallenge(challenge);
```

Comment:
Good practice. Frontendul nu lucrează niciodată cu entitățile bazei de date, in felul acesta se asigura integritatea datelor. Prin urmare, ar trebui ca in request body sa primesti alte tipuri se clase care sa aibe structura asemănătoare entității. Aceste tipuri de clase se numesc DTO- data transfer obiect. Te rog sa citesti despre ele si sa încerci sa actualizezi implementarea.